File: python/processdata.py
#!/usr/bin/python

import cgi, cgitb
import time
import random

cgitb.enable()
print 'Content-Type: text/html;charset=utf-8'
print
print '<!DOCTYPE html>'
print '<html>'

print '<head><title>Form from Python</title></head>'
print '<body>'
form = cgi.FieldStorage()
try:
    username = form['username'].value
    pwd = form['password'].value
    magic = int(form['magicnum'].value)

    for i in range(magic):
        print '<h1>Hello ' + username + ' with a password of ' + pwd + '!</h1>'
except:
    print "<h1>Error</h1>"
    print "Please fill in the form."
    

print '</body></html>'

File: ruby/processdata.rb
#!/usr/bin/ruby
require 'cgi'

cgi = CGI.new

puts "Content-type:text/html\r\n\r\n"
puts '<!DOCTYPE html>'
puts '<html lang="en">'
puts '<head>'
puts '<meta charset="UTF-8">'
puts '<title>Processed with Ruby</title>'
puts '</head>'
puts '<body>'

numLoop = cgi['magicnum']
i=0
while i<numLoop.to_i do
    puts "<h1> Hello #{cgi['username']} with a password of #{cgi['password']}!</h1>"
    i+=1
end
puts '</body>'
puts '</html>'
